examples/tutorials/advanced/focal_mechanisms.py

```diff
@@ -0,0 +1,335 @@
+
+fig = pygmt.Figure()
+fig.basemap(region=region, projection=projection, frame=frame)
+
+for component, longitude in zip(["full", "dc", "deviatoric"], [-3, 0, 2], strict=True):
+    fig.meca(
+        spec=mt_single,
+        scale="1c",
+        longitude=longitude,
+        latitude=0,
+        depth=0,
+        component=component,
+    )
+
+fig.show()
+
+
+# %%
+# Filling the quadrants
+# ---------------------
+#
+# Use the parameters ``compressionfill`` and ``extensionfill`` to fill the quadrants
+# with different colors or patterns. Regarding patterns see the gallery example
+# :doc:`Bit and hachure patterns </gallery/symbols/patterns>` and the Technical
+# Reference :doc:`Bit and hachure patterns </techref/patterns>`.
+
+fig = pygmt.Figure()
+fig.basemap(region=region, projection=projection, frame=frame)
+
+fig.meca(
+    spec=mt_single,
+    scale="1c",
+    longitude=-2,
+    latitude=0,
+    depth=0,
+    compressionfill="darkorange",
+    extensionfill="cornsilk",
+)
+
+fig.meca(
+    spec=mt_single,
+    scale="1c",
+    longitude=2,
+    latitude=0,
+    depth=0,
+    compressionfill="p8",
+    extensionfill="p31",
+    outline=True,
+)
+
+fig.show()
+
+
+# %%
+# Adjusting the outlines
+# ----------------------
+#
+# Use the parameters ``pen`` and ``outline`` for adjusting the circumference of the
+# beachball or all lines (circumference of the beachball and both nodal planes).
+
+fig = pygmt.Figure()
+fig.basemap(region=region, projection=projection, frame=frame)
+
+fig.meca(
+    spec=aki_single,
+    scale="1c",
+    longitude=-2,
+    latitude=0,
+    depth=0,
+    # Use a 1-point thick, darkorange and solid line
+    pen="1p,darkorange",
+)
+
+fig.meca(
+    spec=aki_single,
+    scale="1c",
+    longitude=2,
+    latitude=0,
+    depth=0,
+    outline="1p,darkorange",
+)
+
+fig.show()
+
+
+# %%
+# Highlighting the nodal planes
+# -----------------------------
+#
+# Use the parameter ``nodal``, whereby ``"0"`` refers to both, ``"1"`` to the first, and
+# ``"2"`` to the second nodal plane(s). Only the circumference and the specified nodal
+# plane(s) are plotted, i.e. the quadrants remain unfilled (transparent). If needed,
+# make usage of the stacking concept of (Py)GMT and use ``nodal`` with the ``outline``
+# or / and ``pen``  parameters in combination.
+
+fig = pygmt.Figure()
+fig.basemap(region=region, projection=projection, frame=frame)
+
+fig.meca(
+    spec=aki_single,
+    scale="1c",
+    longitude=-2,
+    latitude=0,
+    depth=0,
+    nodal="0/1p,black",
+)
+
+fig.meca(
+    spec=aki_single,
+    scale="1c",
+    longitude=2,
+    latitude=0,
+    depth=0,
+    compressionfill="lightorange",
+    outline="0.5p,black",
+)
+fig.meca(
+    spec=aki_single,
+    scale="1c",
+    longitude=2,
+    latitude=0,
+    depth=0,
+    nodal="1/1p,darkorange",
+)
+fig.meca(
+    spec=aki_single,
+    scale="1c",
+    longitude=2,
+    latitude=0,
+    depth=0,
+    compressionfill="white@100",
+    extensionfill="white@100",
```

Comment:
As mentioned the suggstions below, it's possible to use `"-"` to mean no filling.